DataAccess/Repository/OrderRepository.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using DataAccess.Data;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+
+namespace DataAccess.Repository
+{
+    public class OrderRepository : IOrderRepository
+    {
+        private BarDataBase _barDataBase;
+
+        public OrderRepository()
+        {
+            _barDataBase = new BarDataBase();
+        }
+
+        public async Task<Order> Get(int id)
+        {
+            Order order = await _barDataBase.Orders.FindAsync(id);
+            if(order == null)
+            {
+                throw new ArgumentException($"No order with {id} was found");
+            }
+            else
+            {
+                return order;
+            }
+        }
+
+        public async Task<int> Upsert(Order order)
+        {
+            bool ord = _barDataBase.Orders.Contains(order);
+            int affrcted;
+            if (ord == false)
+            {
+                _barDataBase.Orders.Add(order);
+                affrcted = await _barDataBase.SaveChangesAsync();
+                return (affrcted == 1) ? -201 : affrcted;
```

Comment:
I see code smell here:
https://blog.devgenius.io/code-smell-02-constants-and-magic-numbers-d6c320eef90b

what is -201, -204? Are these error codes?

Repository should just tell if operation was successful and if not - what was the error. It should not hint towards the type of success :) 
If you do need to differentiate if it was an Update or Insert -> you cant use Upsert.